tests/setup/auth.setup.ts

```diff
@@ -0,0 +1,25 @@
+import { Page } from "@playwright/test";
+import { test as setup, expect, authFile } from "../fixtures/lxd-test";
+
+const loginUser = async (page: Page) => {
+  await page.getByRole("link", { name: "Login with SSO" }).click();
+  await page.getByLabel("Email address*").click();
+  await page.getByLabel("Email address*").fill(process.env.LXD_OIDC_USER || "");
+  await page.getByLabel("Password*").click();
+  await page.getByLabel("Password*").fill(process.env.LXD_OIDC_PASSWORD || "");
+  await page.getByRole("button", { name: "Continue", exact: true }).click();
+  await expect(page.getByText("Log out")).toBeVisible();
+};
+
+setup("authenticate", async ({ page }) => {
+  await page.goto("/ui/");
+  await loginUser(page);
+  // Check logout functionality
+  await page.getByText("Log out").click();
+  await expect(
+    page.getByRole("link", { name: "Login with SSO" }),
+  ).toBeVisible();
+  await loginUser(page);
```

Comment:
Should this be a test case? A bit surprising to log in and log out during setup phase.

Reply:
Yeah I agree it's a little strange. Performing a logout in an underlying test would affect the global auth state since the session id would change. I can maybe try splitting the logout into a separate test but in the same file?

Reply:
Why can't we have a "normal" test that performs log out and log in again?

Reply:
The idea is that this file should be run first before any other tests to setup the global auth file, which would contain session details for an authenticated identity. The dependency is specified in `playwright.config.ts` where the setup file is run in a separate project, prior to any other projects. What I am proposing is a "normal" test but still run inside the setup file, since a logout test would alter the authentication state and should be run before other tests.

Reply:
There is a way to [control test order](https://playwright.dev/docs/next/test-parallel#control-test-order) in CI, as playwright runs tests not in parallel and in alphabetical order. So the logout/login test could be the last one maybe?

Reply:
As discussed, this is a "normal" playwright test file, which is executed first before the other tests. I was under the wrong assumption this is a special setup file, which is not a test. To conclude this can stay as is, all good.